rxbinding-support-v4/src/androidTest/java/com/jakewharton/rxbinding/support/v4/widget/RxSwipeRefreshLayoutTestActivity.java

```diff
@@ -18,15 +21,21 @@
   private final Runnable stopRefreshing = new Runnable() {
     @Override public void run() {
       swipeRefreshLayout.setRefreshing(false);
-      handler.postDelayed(this, 500);
     }
   };
 
   @Override protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
 
     swipeRefreshLayout = new SwipeRefreshLayout(this);
-    swipeRefreshLayout.setId(1);
+    swipeRefreshLayout.setId(R.id.swipe_refresh_layout);
+    swipeRefreshLayout.setOnTouchListener(new View.OnTouchListener() {
+      @Override public boolean onTouch(View v, MotionEvent event) {
+        handler.removeCallbacks(stopRefreshing);
```

Comment:
What if we only did all this on `ACTION_UP`? That would let keep a short delay like 200ms, right?